### PR TITLE
[DOCS] 2D joints docs

### PR DESCRIPTION
**Pull request by Grosskopf**
Added documentation for the 2D joints, mostly copy&paste.

### CODE REVIEW

File: doc/classes/DampedSpringJoint2D.xml [hunk 1]
```diff
@@ -78,12 +78,16 @@
 	</methods>
 	<members>
 		<member name="damping" type="float" setter="set_damping" getter="get_damping">
+			The damping ratio of the spring joint. A value of 0 indicates an undamped spring, while 1 causes the system to reach equilibrium as fast as possible (critical damping).
 		</member>
 		<member name="length" type="float" setter="set_length" getter="get_length">
+			The maximum length of the spring joint.
```
Comment (cbscribe) [inline doc/classes/DampedSpringJoint2D.xml]: As above

File: doc/classes/GrooveJoint2D.xml [hunk 2]
```diff
@@ -46,8 +46,10 @@
 	</methods>
 	<members>
 		<member name="initial_offset" type="float" setter="set_initial_offset" getter="get_initial_offset">
+			The initial offset of the groove on body A.
```
Comment (cbscribe) [inline doc/classes/GrooveJoint2D.xml]: "The groove's initial offset..."

File: doc/classes/Joint2D.xml [hunk 3]
```diff
@@ -74,12 +74,16 @@
 	</methods>
 	<members>
 		<member name="bias" type="float" setter="set_bias" getter="get_bias">
+			The force with which nodes A and B stay in positional relation to each other.
 		</member>
 		<member name="disable_collision" type="bool" setter="set_exclude_nodes_from_collision" getter="get_exclude_nodes_from_collision">
+			Decides, if [member node_a] and [member node_b] should be able to collide.
```
Comment (cbscribe) [inline doc/classes/Joint2D.xml]: See [Writing Guidelines](http://docs.godotengine.org/en/latest/community/contributing/docs_writing_guidelines.html): boolean members have a standard wording.  Finally, it's helpful to show the default value:

    If [code]true[/code] [member node_a] and [member node_b] can collide. Default value: [code]false[/code].

File: doc/classes/Joint2D.xml [hunk 4]
```diff
@@ -74,12 +74,16 @@
...
 		</member>
 		<member name="node_a" type="NodePath" setter="set_node_a" getter="get_node_a">
+			The A node for the joint. Must be of type [PhysicsBody2D].
```
Comment (cbscribe) [inline doc/classes/Joint2D.xml]: To be more accurate, the node's type has to be one that inherits PhysicsBody2D.

    The joint's A node. Must be a [PhysicsBody2D] derived node.

File: doc/classes/Joint2D.xml [hunk 5]
```diff
@@ -74,12 +74,16 @@
...
 		</member>
 		<member name="node_b" type="NodePath" setter="set_node_b" getter="get_node_b">
+			The B node for the joint. Must be of type [PhysicsBody2D].
```
Comment (cbscribe) [inline doc/classes/Joint2D.xml]: As above.

File: doc/classes/Joint2D.xml [hunk 6]
```diff
@@ -74,12 +74,16 @@
 	</methods>
 	<members>
 		<member name="bias" type="float" setter="set_bias" getter="get_bias">
+			The force with which nodes A and B stay in positional relation to each other.
```
Comment (cbscribe) [inline doc/classes/Joint2D.xml]: This needs more description. Is the force constant or proportional to the distance? "positional relation" sounds awkward, how about "the two nodes' relative positions"?

File: doc/classes/GrooveJoint2D.xml [hunk 7]
```diff
@@ -46,8 +46,10 @@
...
 		</member>
 		<member name="length" type="float" setter="set_length" getter="get_length">
+			The length of the groove.
```
Comment (cbscribe) [inline doc/classes/GrooveJoint2D.xml]: The groove's length.

File: doc/classes/DampedSpringJoint2D.xml [hunk 8]
```diff
@@ -78,12 +78,16 @@
...
 		</member>
 		<member name="rest_length" type="float" setter="set_rest_length" getter="get_rest_length">
+			The resting length of the spring joint. The joint will always try to go to back this length when pulled apart.
```
Comment (cbscribe) [inline doc/classes/DampedSpringJoint2D.xml]: The joint's resting length. 

Does the joint not return to this length if it's compressed?

File: doc/classes/DampedSpringJoint2D.xml [hunk 9]
```diff
@@ -78,12 +78,16 @@
 	</methods>
 	<members>
 		<member name="damping" type="float" setter="set_damping" getter="get_damping">
+			The damping ratio of the spring joint. A value of 0 indicates an undamped spring, while 1 causes the system to reach equilibrium as fast as possible (critical damping).
```
Comment (cbscribe) [inline doc/classes/DampedSpringJoint2D.xml]: Per Writing Guidelines, avoid the "of the" construction - it just leads to longer sentences. Instead use "The spring joint's damping ratio."

You need to indicate what "undamped spring" means.

**Review by cbscribe — CHANGES_REQUESTED** [review]

File: doc/classes/DampedSpringJoint2D.xml (hunk 9 above)
Comment (Grosskopf) [reply doc/classes/DampedSpringJoint2D.xml]: oh you're right, thanks :)

File: doc/classes/DampedSpringJoint2D.xml [hunk 10]
```diff
@@ -78,12 +78,16 @@
 	</methods>
 	<members>
 		<member name="damping" type="float" setter="set_damping" getter="get_damping">
+			The spring joint's damping ratio. A value of 0 indicates an undamped spring, while 1 causes the system to reach equilibrium as fast as possible (critical damping).
```
Comment (cbscribe) [inline doc/classes/DampedSpringJoint2D.xml]: My other comment still applies: there needs to be an explanation of what an "undamped spring" is, something along the lines of "the higher the value, the faster the spring reaches equilibrium."

File: doc/classes/Joint2D.xml [hunk 11]
```diff
@@ -74,12 +74,16 @@
 	</methods>
 	<members>
 		<member name="bias" type="float" setter="set_bias" getter="get_bias">
+			The force with which nodes A and B stay in the same relative position to the joint over time.
```
Comment (cbscribe) [inline doc/classes/Joint2D.xml]: This is still a really difficult sentence to parse.  I don't really have any idea what's going on with this property.

File: doc/classes/DampedSpringJoint2D.xml (hunk 10 above)
Comment (Grosskopf) [reply doc/classes/DampedSpringJoint2D.xml]: oh ok

File: doc/classes/Joint2D.xml (hunk 11 above)
Comment (Grosskopf) [reply doc/classes/Joint2D.xml]: ok, will explain more :)

File: doc/classes/DampedSpringJoint2D.xml [hunk 12]
```diff
@@ -78,12 +78,18 @@
 	</methods>
 	<members>
 		<member name="damping" type="float" setter="set_damping" getter="get_damping">
+			The spring joint's damping ratio.
+			When the two bodies move into different directions the system tries to allign them to the spring axis again.
```
Comment (NathanLovato) [inline doc/classes/DampedSpringJoint2D.xml]: 3 lines = 3 paragraphs in the docs. One l for align and alignment. Suggestion:

The spring joint's damping ratio. A value between [code]0[/code] and [code]1[/code]. When the two bodies move into different directions the system tries to align them to the spring axis again. A high [code]damping[/code] value forces the attached bodies to align faster. Default value: [code]add default[/code]

File: doc/classes/DampedSpringJoint2D.xml [hunk 13]
```diff
@@ -78,12 +78,18 @@
...
+			This value defines how fast this happens, a higher value makes the allignment happen faster.
 		</member>
 		<member name="length" type="float" setter="set_length" getter="get_length">
+			The spring joint's maximum length.
```
Comment (NathanLovato) [inline doc/classes/DampedSpringJoint2D.xml]: `The spring joint's maximum length. The two attached bodies cannot stretch it past this value. Default value: [code]add default[/code]`

File: doc/classes/DampedSpringJoint2D.xml [hunk 14]
```diff
@@ -78,12 +78,18 @@
...
 		</member>
 		<member name="rest_length" type="float" setter="set_rest_length" getter="get_rest_length">
+			The spring joint's resting length. The joint will always try to go to back this length when pulled apart or pushed together.
```
Comment (NathanLovato) [inline doc/classes/DampedSpringJoint2D.xml]: `When the bodies attached to the spring joint move they stretch or squash it. The joint always tries to resize towards this length. Default value: [code]add default[/code]`

File: doc/classes/DampedSpringJoint2D.xml [hunk 15]
```diff
@@ -78,12 +78,18 @@
...
 		</member>
 		<member name="stiffness" type="float" setter="set_stiffness" getter="get_stiffness">
+			The joint applies a force equal to the stiffness times the distance from its resting length.
```
Comment (NathanLovato) [inline doc/classes/DampedSpringJoint2D.xml]: Unclear.
`The higher the value, the less the bodies attached to the joint will deform it. The joint applies an opposing force to the bodies, the product of the stiffness multiplied by the size difference from its resting length. Default value: ...`

File: doc/classes/GrooveJoint2D.xml [hunk 16]
```diff
@@ -46,8 +46,10 @@
 	</methods>
 	<members>
 		<member name="initial_offset" type="float" setter="set_initial_offset" getter="get_initial_offset">
+			The grooves initial offset of the on body A.
```
Comment (NathanLovato) [inline doc/classes/GrooveJoint2D.xml]: Needs a rewrite. It's the starting distance to the first body attached to the joint?

File: doc/classes/GrooveJoint2D.xml [hunk 17]
```diff
@@ -46,8 +46,10 @@
...
 		</member>
 		<member name="length" type="float" setter="set_length" getter="get_length">
+			The groove's length.
```
Comment (NathanLovato) [inline doc/classes/GrooveJoint2D.xml]: What does it do? I can tell from the variable name it's the joint's length. Also, you wrote "spring joint" for the sprint joint class: how about using groove joint here?

File: doc/classes/Joint2D.xml [hunk 18]
```diff
@@ -74,12 +74,18 @@
 	</methods>
 	<members>
 		<member name="bias" type="float" setter="set_bias" getter="get_bias">
+			Currently only used in [PinJoint2D].
+			When [member node_a] and [member node_b] move in different directions, this value defines how much they get pulled back into the same positional relation to each other.
```
Comment (NathanLovato) [inline doc/classes/Joint2D.xml]: Merge the 2 paragraphs. "positional relation to each other" lacks clarity. Another passive. You have a link to this variable.
`When [member node_a] and [member node_b] move in different directions the [code]bias[/code] controls how fast the joint pulls them back to their original position. The lower the [code]bias[/code] the more the two bodies can pull on the joint. Default value: ...`

File: doc/classes/Joint2D.xml [hunk 19]
```diff
@@ -74,12 +74,18 @@
...
+			The lower the [member bias], the joints' attachment to each other can flex more.
 		</member>
 		<member name="disable_collision" type="bool" setter="set_exclude_nodes_from_collision" getter="get_exclude_nodes_from_collision">
+			If [code]true[/code] [member node_a] and [member node_b] can collide. Default value: [code]false[/code].
 		</member>
 		<member name="node_a" type="NodePath" setter="set_node_a" getter="get_node_a">
+			The joint's A node. Must be a [PhysicsBody2D] derived node.
```
Comment (NathanLovato) [inline doc/classes/Joint2D.xml]: `The first body attached to the joint. Must derive from [PhysicsBody2D].`

File: doc/classes/Joint2D.xml [hunk 20]
```diff
@@ -74,12 +74,18 @@
...
 		</member>
 		<member name="node_b" type="NodePath" setter="set_node_b" getter="get_node_b">
+			The joint's B node. Must be a [PhysicsBody2D] derived node.
```
Comment (NathanLovato) [inline doc/classes/Joint2D.xml]: `The second body attached to the joint. Must derive from [PhysicsBody2D].`

**Review by NathanLovato — CHANGES_REQUESTED** [review]
Some descriptions are hard to understand. I left some suggestions you can use as is to update faster.
Missing default values
Passive voice

**Comment by Grosskopf** [comment]
Sorry for the terrible documentation, i shouldn't have copied and pasted the old stuff quick and dirty :/

**Review by NathanLovato — APPROVED** [review]

**Comment by NathanLovato** [comment]
Good job, no worries! Your work is always much appreciated

**Pull request merged**